File: application/backend-credit-card-enrollment/backend-c#/Common/Query/QueryHandler.cs
using CreditCardEnrollment.Common.Projection;

namespace CreditCardEnrollment.Common.Query;

public abstract class QueryHandler {
    protected readonly MongoTransactionalProjectionOperator _mongoTransactionalProjectionOperator;

    protected QueryHandler(MongoTransactionalProjectionOperator mongoTransactionalProjectionOperator) {
        _mongoTransactionalProjectionOperator = mongoTransactionalProjectionOperator;
    }

    public abstract object HandleQuery(Query query);
}

File: application/backend-credit-card-enrollment/backend-typescript/src/common/query/QueryHandler.ts
import { MongoTransactionalProjectionOperator } from '../projection/MongoTransactionalProjectionOperator';
import { Query } from './Query';

export abstract class QueryHandler {
    constructor(
        protected readonly mongoTransactionalProjectionOperator: MongoTransactionalProjectionOperator
    ) {}

    abstract handleQuery(query: Query): Promise<unknown>;
}